client/sites/settings/agency/index.jsx

```diff
@@ -1,7 +1,58 @@
-export default function AgencySettings() {
+import { __ } from '@wordpress/i18n';
+import useFetchAgencyFromBlog from 'calypso/a8c-for-agencies/data/agencies/use-fetch-agency-from-blog';
+import NavigationHeader from 'calypso/components/navigation-header';
+import { A4AFullyManagedSiteSetting } from 'calypso/my-sites/site-settings/a4a-fully-managed-site-setting';
+import wrapSettingsForm from 'calypso/my-sites/site-settings/wrap-settings-form';
+import { useSelector } from 'calypso/state';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+const AgencySettings = ( {
+	fields,
+	handleSubmitForm,
+	handleToggle,
+	isRequestingSettings,
+	isSavingSettings,
+} ) => {
+	const site = useSelector( getSelectedSite );
+	const isAtomicSite = site.is_wpcom_atomic;
+
+	const { data: agencySite } = useFetchAgencyFromBlog( site?.ID, { enabled: !! site?.ID } );
+
+	const shouldShowToggle = agencySite && isAtomicSite;
```

Comment:
Hmm... I think we need to start thinking about this redundant guard. It makes the code a bit convoluted just to accommodate a very edge case. The page will be broken anyway due to the lack of active sidebar item.

I think we can do something like 

```jsx
<SidebarItem enabled={shouldShowAgency} ...>
```

and in `SidebarItem`, we just throw 404 / redirect to /overview. What do you think?

Reply:
Proposing: https://github.com/Automattic/wp-calypso/pull/95937

Reply:
Left some comments, agree there's an issue here not quite sure what the best approach is yet.